src/main/scala/cs332/distributedsorting/slave/Slave.scala

```diff
@@ -1,51 +1,99 @@
 package cs332.distributedsorting.slave
 
+import cs332.distributedsorting.common.Util.getMyIpAddress
+import scala.io.Source 
 import java.util.concurrent.TimeUnit
 import java.util.logging.{Level, Logger}
 import io.grpc.{ManagedChannel, ManagedChannelBuilder, StatusRuntimeException}
-
-import cs332.distributedsorting.example.{GreeterGrpc, HelloRequest}
-import cs332.distributedsorting.example.GreeterGrpc.GreeterBlockingStub
+import cs332.distributedsorting.sorting.{HandshakeRequest, SortingGrpc, SendDataRequest, SendDataResponse}
+import cs332.distributedsorting.sorting.SortingGrpc.SortingBlockingStub
 
 object Slave {
   def apply(host: String, port: Int): Slave = {
     val channel = ManagedChannelBuilder.forAddress(host, port).usePlaintext().asInstanceOf[ManagedChannelBuilder[_]].build
-    val blockingStub = GreeterGrpc.blockingStub(channel)
+    val blockingStub = SortingGrpc.blockingStub(channel)
     new Slave(channel, blockingStub)
   }
 
   def main(args: Array[String]): Unit = {
-    val client = Slave("localhost", 50051)
-    try {
-      val user = args.headOption.getOrElse("world")
-      client.greet(user)
-    } finally {
-      client.shutdown()
+    val masterEndpoint = args.headOption
+    if (masterEndpoint.isEmpty)
+      System.out.println("Master ip:port argument is empty.")
+    else {
+      System.out.println("Try to handshake with master: " + masterEndpoint.get)
+      val splitedEndpoint = masterEndpoint.get.split(':')
+      val client = Slave(splitedEndpoint(0), splitedEndpoint(1).toInt)
+      try {
+        client.handshake()
+      }
+      catch{
+        case e : Exception => {
+          client.shutdown()
+          return
+        }
+      }
+        // suppose args(1) is  the path to data file generated with gensort
+      val path = args.lastOption
+      if (path.isEmpty){
+        System.out.println("Path to data file is empty.")
+        client.shutdown()
+      }
+      else{
+        System.out.println("Try to send data to master")
+        val fSource = Source.fromFile(args(0))
+        val keyList = (fSource.grouped(100).map(x=>x.dropRight(90))).take(100000).toString() // list of key (size is 1GB)
```

Comment:
I think 1GB is too big for each file. Maybe it will be enough for 1MB per file for sampling data.

Reply:
I did not test with real gensort data yet. I will add this test to my weekly task. And for the size of sampling data, I change for 1MB, but if it 's not enough it will be very easy to increase it after the tests.

Reply:
ok, than you can add commits for the test code of creating partition in this pull request.